CRM/Contribute/BAO/Contribution.php

```diff
@@ -493,91 +497,45 @@ public static function create(&$params, $ids = []) {
     }
 
     $params['contribution_id'] = $contribution->id;
-    $session = CRM_Core_Session::singleton();
 
     if (!empty($params['note'])) {
       $noteParams = [
         'entity_table' => 'civicrm_contribution',
         'note' => $params['note'],
         'entity_id' => $contribution->id,
-        'contact_id' => $session->get('userID'),
+        'contact_id' => CRM_Core_Session::getLoggedInContactID(),
       ];
       if (!$noteParams['contact_id']) {
         $noteParams['contact_id'] = $params['contact_id'];
       }
       CRM_Core_BAO_Note::add($noteParams);
     }
 
-    // make entry in batch entity batch table
-    if (!empty($params['batch_id'])) {
-      // in some update cases we need to get extra fields - ie an update that doesn't pass in all these params
-      $titleFields = [
-        'contact_id',
-        'total_amount',
-        'currency',
-        'financial_type_id',
-      ];
-      $retrieveRequired = 0;
-      foreach ($titleFields as $titleField) {
-        if (!isset($contribution->$titleField)) {
-          $retrieveRequired = 1;
-          break;
-        }
-      }
-      if ($retrieveRequired == 1) {
-        $contribution->find(TRUE);
-      }
-    }
```

Comment:
@mattwire why on earth do we do this if batch_id is present? It looks to me an awful lot like logic to support a specific form layer flow - I just added https://github.com/civicrm/civicrm-core/pull/17120 to figure out if it is tested at any point if so where it is hit from - I'm guessing we can likely remove this

Reply:
Looks like it does get hit at least in one place https://test.civicrm.org/job/CiviCRM-Core-PR/33520/testReport/junit/(root)/CRM_Contribute_Form_SearchTest/testBatchFilter/

Reply:
Yep  - it's not clear the loading that is happening adds any value though!